sources/reference/tutorials/batch.md

```diff
@@ -0,0 +1,606 @@
+        ]
+    },
+    "method": [],
+    "configuration": {
+        "aws_region": "<REGION>"
+    }
+}
+```
+
+Replace the value of `<INSTANCE_ID>` with the value of the id of the deployed
+instance. Replace `<REGION>` with the name of the region the instance is deployed
+in.
+
+### 3. Setup your Docker image
+
+AWS Batch requires us to have a Docker image within the Job definition, this
+container will be _what_ does the work for our Batch Job.
+
+Make a file named `Dockerfile` alongside `experiments/` with the following
+contents:
+
+```Dockerfile
+FROM chaostoolkit/chaostoolkit:latest
+
+RUN pip install chaostoolkit-aws
+
+RUN mkdir /home/svc/experiments
+
+COPY experiments /home/svc/experiments
+
+WORKDIR /home/svc/experiments
+```
+
+### 4. Create your ECR repository and push the image
+
+* Navigate to the ECR console and select `Repositories`
+* Select `Create Repository`
+* Leave the repository set to `Private` and enter a name for your repository
+    * For the purpose of this guide, we'll be using `ctk-batch`
+* Select `Create Repository`
+* Select `<repository-name>` from the table
+* Select `View push commands`
+* Follow the commands outlined there (We'll show you them below as an example)
+
+**Logging in to ECR with Docker**
+```console
+aws ecr get-login-password --region eu-west-2 | docker login --username AWS --password-stdin <your-aws-account-id>.dkr.ecr.eu-west-2.amazonaws.com
+```
+
+```console
+Login Succeeded
+```
+
+**Building the image**
+```console
+docker build -t ctk-batch .
+```
+
+```console
+[+] Building 1.8s (10/10) FINISHED
+ => [internal] load build definition from Dockerfile        0.0s
+ => => transferring dockerfile: 220B                        0.0s
+ => [internal] load .dockerignore                           0.0s
+ => => transferring context: 2B                             0.0s
+ => [internal] load metadata for docker.io/chaostoolkit/chaostoolkit:latest                                                         1.5s
+ => [1/5] FROM docker.io/chaostoolkit/chaostoolkit:latest@sha256:3801eda37de7e8f00fb556220fff7935fea45d248881f4253cd9c29b4d3023f3   0.0s
+ => => resolve docker.io/chaostoolkit/chaostoolkit:latest@sha256:3801eda37de7e8f00fb556220fff7935fea45d248881f4253cd9c29b4d3023f3   0.0s
+ => [internal] load build context                           0.0s
+ => => transferring context: 959B                           0.0s
+ => CACHED [2/5] RUN pip install chaostoolkit-aws           0.0s
+ => CACHED [3/5] RUN mkdir /home/svc/experiments            0.0s
+ => [4/5] COPY experiments /home/svc/experiments            0.0s
+ => [5/5] WORKDIR /home/svc/experiments                     0.0s
+ => exporting to image                                      0.1s
+ => => exporting layers                                     0.0s
+ => => writing image sha256:4a3ce8f2824518bffa47ff3d293488f18f83e25711bedc32e13611a5c7e7e0af                                        0.0s
+ => => naming to docker.io/library/ctk-batch                0.0s
```

Comment:
This is the output after the docker build has already been created and has been cached
A first time run looks like this - if it matters:

[+] Building 10.7s (10/10) FINISHED                                                                                 
 => [internal] load build definition from Dockerfile                                                           0.0s
 => => transferring dockerfile: 37B                                                                            0.0s
 => [internal] load .dockerignore                                                                              0.0s
 => => transferring context: 2B                                                                                0.0s
 => [internal] load metadata for docker.io/chaostoolkit/chaostoolkit:latest                                    1.5s
 => [1/5] FROM docker.io/chaostoolkit/chaostoolkit:latest@sha256:3801eda37de7e8f00fb556220fff7935fea45d248881  2.6s
 => => resolve docker.io/chaostoolkit/chaostoolkit:latest@sha256:3801eda37de7e8f00fb556220fff7935fea45d248881  0.0s
 => => sha256:dbd1ca4c69c528490843f7631a396846bbf486bf35e67afa5d9fc06ebbde39c4 9.64kB / 9.64kB                 0.0s
 => => sha256:3801eda37de7e8f00fb556220fff7935fea45d248881f4253cd9c29b4d3023f3 1.99kB / 1.99kB                 0.0s
 => => sha256:29291e31a76a7e560b9b7ad3cada56e8c18d50a96cca8a2573e4f4689d7aca77 2.81MB / 2.81MB                 0.6s
 => => sha256:3626a090061d56f9d6eaa946c29f4a7600401e2bca3f7b517369d0da709bf6d3 281.50kB / 281.50kB             0.3s
 => => sha256:c6625ad8c0ff0a54d772b65229073373de275c2f7a16c71cbbf584ca1185d0ce 10.20MB / 10.20MB               1.2s
 => => sha256:5b904a152bb436e642d2db20578ff1b60d5f47481c2bc87ca9702c57cca61ba5 231B / 231B                     0.6s
 => => extracting sha256:29291e31a76a7e560b9b7ad3cada56e8c18d50a96cca8a2573e4f4689d7aca77                      0.2s
 => => sha256:916e8b567dc502c85c84849d3d41273c3188ad6f7b66de7819de462b2a771772 2.21MB / 2.21MB                 1.0s
 => => sha256:ad48bf7470a4164485c362c3ac0a53ac7496ebb96b7f42b8655fd1d30d5d23c9 6.24MB / 6.24MB                 1.6s
 => => extracting sha256:3626a090061d56f9d6eaa946c29f4a7600401e2bca3f7b517369d0da709bf6d3                      0.1s
 => => sha256:d5a998bb98a93f93c72f33fecb9eed51c82a88a421345f42140f84b51547afc8 547B / 547B                     1.2s
 => => sha256:8c142b52e89fce4dbacef1fde29dbe956da9f654301084905491e1b855ce7177 1.26kB / 1.26kB                 1.4s
 => => extracting sha256:c6625ad8c0ff0a54d772b65229073373de275c2f7a16c71cbbf584ca1185d0ce                      0.4s
 => => extracting sha256:5b904a152bb436e642d2db20578ff1b60d5f47481c2bc87ca9702c57cca61ba5                      0.0s
 => => extracting sha256:916e8b567dc502c85c84849d3d41273c3188ad6f7b66de7819de462b2a771772                      0.2s
 => => extracting sha256:ad48bf7470a4164485c362c3ac0a53ac7496ebb96b7f42b8655fd1d30d5d23c9                      0.3s
 => => extracting sha256:d5a998bb98a93f93c72f33fecb9eed51c82a88a421345f42140f84b51547afc8                      0.0s
 => => extracting sha256:8c142b52e89fce4dbacef1fde29dbe956da9f654301084905491e1b855ce7177                      0.0s
 => [internal] load build context                                                                              0.0s
 => => transferring context: 7.27kB                                                                            0.0s
 => [2/5] RUN pip install chaostoolkit-aws                                                                     5.7s
 => [3/5] RUN mkdir /home/svc/experiments                                                                      0.3s
 => [4/5] COPY experiments /home/svc/experiments                                                               0.0s 
 => [5/5] WORKDIR /home/svc/experiments                                                                        0.0s 
 => exporting to image                                                                                         0.5s 
 => => exporting layers                                                                                        0.5s 
 => => writing image sha256:02b2de1e0f4b75fa6a39a8b643a8cab3713782d0f1bca8a4dc42919c2430f27c                   0.0s 
 => => naming to docker.io/library/ctk-batch                                                                   0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them

Reply:
Yeah, tbh I'm not even sure how useful outputs are, but as long as it ran OK, that's the important bit haha

Reply:
coolio, just wanted to check